### PR TITLE
Complete daniel-chambers' PR adding Show instances to generics rep types

### PR DESCRIPTION
**Pull request by JordanMartinez**
Completes the PR he started in https://github.com/purescript/purescript-generics-rep/pull/11

### CODE REVIEW

File: test/Data/Generic/Rep.purs [hunk 1]
```diff
@@ -123,6 +123,9 @@ testGenericRep = do
   assert "Checking show" $
     show (cons 1 (cons 2 Nil)) == "(Cons { head: 1, tail: (Cons { head: 2, tail: Nil }) })"
 
+  assert "Checking show for generic types" $
+    show (G.from $ cons 1 Nil) == "(Inr (Constructor \"Cons\" (Argument { head: 1, tail: Nil })))"
+
```
Comment (kl0tl) [inline test/Data/Generic/Rep.purs]: Could we also test the show output for `G.from Nil` to cover `NoArguments` and the `Inl` constructor, as well as a constructor with multiple arguments to cover `Product`?

File: src/Data/Generic/Rep.purs [hunk 2]
```diff
@@ -10,25 +10,46 @@ module Data.Generic.Rep
...
 
+instance showProduct :: (Show a, Show b) => Show (Product a b) where
+  show (Product a b) = "(Product " <> show a <> " " <> show b <> ")"
+
 -- | A representation for constructors which includes the data constructor name
 -- | as a type-level string.
 newtype Constructor (name :: Symbol) a = Constructor a
 
+instance showConstructor :: (IsSymbol name, Show a) => Show (Constructor name a) where
+  show (Constructor a) = "(Constructor " <> show (reflectSymbol (Proxy :: Proxy name)) <> " " <> show a <> ")"
```
Comment (hdgarrood) [inline src/Data/Generic/Rep.purs]: Can I request that we put a `@` symbol before the constructor name? It wouldn’t be valid syntax (we don’t have explicit type applications) but it does at least clarify that the string isn’t an actual constructor argument for Constructor.

Comment (JordanMartinez) [reply src/Data/Generic/Rep.purs]: Meaning this? Yeah, I think that makes sense.
```diff
- "(Constructor " <> show (reflectSymbol ...
+ "(Constructor @" <> show (reflectSymbol ...
```

Comment (hdgarrood) [reply src/Data/Generic/Rep.purs]: Yeah, exactly that.

File: test/Data/Generic/Rep.purs (hunk 1 above)
Comment (JordanMartinez) [reply test/Data/Generic/Rep.purs]: Yup. Fixed.

File: src/Data/Generic/Rep.purs (hunk 2 above)
Comment (JordanMartinez) [reply src/Data/Generic/Rep.purs]: Fixed.

**Review by hdgarrood — APPROVED** [review]

**Pull request merged**